core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -136,10 +135,14 @@ static inline int ti_sci_get_response(struct ti_sci_xfer *xfer)
 static inline int ti_sci_do_xfer(struct ti_sci_xfer *xfer)
 {
 	struct k3_sec_proxy_msg *msg = &xfer->tx_message;
+	struct ti_sci_msg_hdr *hdr = NULL;
 	int ret = 0;
 
 	mutex_lock(&ti_sci_mutex_lock);
 
+	hdr = (struct ti_sci_msg_hdr *)msg->buf;
+	hdr->seq = ++message_sequence;
```

Comment:
For consistency in OP-TEE OS implementation, prefer a 2 step implementation and let compiler optimize the thing:
```c
message_sequence++;
hdr->seq = message_sequence;